bin/utils.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
```

Comment:
I suggest moving this outside of bin/ it's not an executable. Maybe a new directory lib/src or just lib/?

Reply:
lib/src IMO.